snuba/migrations/snuba_migrations/sessions/0002_sessions_aggregates.py

```diff
@@ -0,0 +1,128 @@
+from typing import Sequence, Tuple
+
+from snuba.clickhouse.columns import (
+    AggregateFunction,
+    Column,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+DEFAULT_QUANTITY = 1
+
+new_raw_columns: Sequence[Tuple[Column[Modifiers], str]] = [
+    (
+        Column("quantity", UInt(32, Modifiers(default=str(DEFAULT_QUANTITY)))),
+        "distinct_id",
+    ),
+    (Column("user_agent", String(Modifiers(low_cardinality=True))), "environment"),
+    (Column("os", String(Modifiers(low_cardinality=True))), "user_agent"),
+]
+
+new_dest_columns: Sequence[Tuple[Column[Modifiers], str]] = [
+    (Column("user_agent", String(Modifiers(low_cardinality=True))), "environment"),
+    (Column("os", String(Modifiers(low_cardinality=True))), "user_agent"),
```

Comment:
I assume the materialized view will group by over these columns. Do you have a plan on how you are going to use them in the produce knowing that old data will have these columns as NULL? (since the old rows are NULL, you will have a big group where os and uuser_agent are not distinguishable. Is that ok ?)

Reply:
yes, that is the intention, and we actually need to figure out how to best show these in the UI